websocket/connection.go

```diff
@@ -130,6 +137,21 @@ func NewDurableConnection(target string, messageChan chan []byte, logger *zap.Su
 		}
 	}()
 
+	// Keep sending pings 3 times per pongTimeout interval.
+	go func() {
+		ticker := time.NewTicker(pongTimeout / 3)
+		for {
+			select {
+			case <-ticker.C:
+				if err := c.write(websocket.PingMessage, []byte{}); err != nil {
+					logger.Errorw("Failed to send ping message", zap.Error(err))
```

Comment:
should this trigger a reconnect?

Reply:
in principle we could make it so. However, then we should also check the errors of usual `Send` calls and trigger reconnects accordingly.

Ultimately, the ping/pong mechanism should catch this reconnect desire in this case. Being quicker on Send errors seems a different can of worms I wouldn't want to open in this PR.